src/EventStore.Core/Certificates/CertificateUtils.cs

```diff
@@ -96,18 +97,41 @@ private static bool TryReadPkcs12CertificateBundle
 				}
 
 				// if the above attempt failed, try to load the certificate via the standard constructor
-				var certificate = new X509Certificate2(certificatePath, password);
+				var certificate = new X509Certificate2(certificatePath, certificatePassword);
 				if (!certificate.HasPrivateKey)
 					throw new NoCertificatePrivateKeyException();
 				return (certificate, null);
 			}
 
-			var privateKey = Convert.FromBase64String(string.Join(string.Empty, File.ReadAllLines(privateKeyPath)
-				.Skip(1)
-				.SkipLast(1)));
+			string[] allLines = File.ReadAllLines(privateKeyPath);
+			var header = allLines[0].Replace("-", "");
+			var privateKey = Convert.FromBase64String(string.Join(string.Empty, allLines.Skip(1).SkipLast(1)));
 
 			using var rsa = RSA.Create();
-			rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+			switch (header) {
+				case "BEGIN PRIVATE KEY":
+					rsa.ImportPkcs8PrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN RSA PRIVATE KEY":
+					rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN ENCRYPTED PRIVATE KEY":
+					if (certificatePrivateKeyPassword is null) {
+						throw new ArgumentException(
+							"A password is required to read the certificate's private key file.");
+					}
+					try {
+						rsa.ImportEncryptedPkcs8PrivateKey(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(certificatePrivateKeyPassword)), new ReadOnlySpan<byte>(privateKey), out _);
+					} catch (CryptographicException e) {
+						throw new ArgumentException(
+							"Failed to read the certificate's private key file. The password may be incorrect.", e);
```

Comment:
```suggestion
							"Failed to read the certificate's private key file. The password may be incorrect or the private key may not be an RSA key.", e);
```

Reply:
@hunter1703 i've tested with an encrypted EC key and it throws the same type of exception. which may be misleading if we don't mention we're also expecting an RSA key.

Reply:
Done. Thanks for the comments

Reply:
@hunter1703 thanks!